### ISSUE TITLE
Add extra trace level logs for tracing

### ISSUE THREAD

**Issue by fommil** [DataDog/datadog-agent#3232]
Please add additional logs when `DD_LOG_LEVEL=trace` is enabled, that details the `span_id` and `trace_id` for submissions, and `parent_id` if it is available, along with confirmation that the trace has been accepted and will be submitted upstream.

At this log level, I'd also like to see the stats information printed after every submission. Or a summary at the end of processing to confirm that a span has not been dropped.

We are currently trying to debug an issue with the Analytics team for our Haskell clietn and we need to prove that the agent is receiving multiple spans for the same trace_id and that those spans are not being dropped.

**Comment by fommil** [comment]
For example, these are some logs from the agent with TRACE logging


```
[ TRACE ] 2019-04-01 16:31:22 TRACE (trace.go:184) - Handling new trace with 1 spans: [service:"node-1" name:"apolloSign" resource:"time" traceID:15107535988506311581 spanID:4802645862389887117 parentID:11805589115589964803 start:1554136282572062657 duration:1245521 meta:<key:"signer" value:"KA-1962364902214711" > metrics:<key:"_sample_rate" value:1 > metrics:<key:"_sampling_priority_v1" value:2 > metrics:<key:"_top_level" value:1 > ]
[ TRACE ] 2019-04-01 16:31:22 TRACE (trace.go:184) - Handling new trace with 1 spans: [service:"node-1" name:"toExecutable_case" resource:"time" traceID:15107535988506311581 spanID:7753572055190672701 parentID:4153416653497057438 start:1554136282591093612 duration:52566 meta:<key:"txindex" value:"560.0" > meta:<key:"txtype" value:"Execute" > metrics:<key:"_sample_rate" value:1 > metrics:<key:"_sampling_priority_v1" value:2 > metrics:<key:"_top_level" value:1 > ]
[ TRACE ] 2019-04-01 16:31:22 TRACE (trace.go:184) - Handling new trace with 1 spans: [service:"node-1" name:"toExecutable" resource:"time" traceID:15107535988506311581 spanID:4153416653497057438 parentID:13828829508253774982 start:1554136282589961856 duration:1554879 metrics:<key:"_sample_rate" value:1 > metrics:<key:"_sampling_priority_v1" value:2 > metrics:<key:"_top_level" value:1 > ]
[ TRACE ] 2019-04-01 16:31:22 TRACE (trace.go:184) - Handling new trace with 1 spans: [service:"node-1" name:"scheduleTransaction" resource:"time" traceID:15107535988506311581 spanID:175758870695230718 parentID:13828829508253774982 start:1554136282591750267 duration:72412706 metrics:<key:"_sample_rate" value:1 > metrics:<key:"_sampling_priority_v1" value:2 > metrics:<key:"_top_level" value:1 > ]
```

But only one of these spans shows up in the UI

https://app.datadoghq.com/apm/trace/15107535988506311581

(this requires permissions to access)

We're trying to find out **why** they don't show up, which may be a bug in the `datadog-agent` or a misunderstanding about the spec on our side. There is currently no logging to indicate that there is a problem.

**Comment by fommil** [comment]
Update: in a slack conversation we found that

1. `parent_id` is required and only one span can have 0 (the root span).
1. we have multiple "root" spans, so the datadog model is problematic.
1. if `parent_id` is omitted, there is no log to indicate that the span has been rejected / dropped.
1. the UI is randomly picking one of these spans to show, we don't know why

**Comment by fommil** [comment]
Also, my spans all have `_top_level`. The documentation in the Go code says

```
	// This is a special metric, it's 1 if the span is top-level, 0 if not.
	topLevelKey = "_top_level"
```

but what does "top level" mean?

**Comment by gbbr** [comment]
> but what does "top level" mean?

Top level means that it is the first span in a hierarchy that belongs to a new service (different than its parent). It's the first span in the service it belongs to.

> if parent_id is omitted, there is no log to indicate that the span has been rejected / dropped (seems to be dropped silently in the Datadog UI).

If you are using `debug` log level you will always see when something is dropped at the agent level.

> the UI is randomly picking one of these spans to show, we don't know why

You should note (not sure if that's the case for you) that in Datadog the operation name (span's "name" field) should be constant for the same code path. Do not use variable values as you would for other instrumentation systems. For variable values use the "resource" field. If you use variable operation names you will see the most used one only in Datadog.

Additionally, if you reach out through support, we can look at your account's org. ID and see more information on what's going on.

**Comment by fommil** [comment]
> Top level means that it is the first span in a hierarchy that belongs to a new service

Hmm, then this is very strange because *all* of my spans are getting assigned to top level.

> You should note (not sure if that's the case for you) that in Datadog the operation name (span's "name" field) should be constant for the same code path. Do not use variable values as you would for other instrumentation systems. For variable values use the "resource" field. If you use variable operation names you will only see the most used one in the UI.

Could you please ellaborate on this? When you say "the same code path" do you mean between versions of my code or do you mean that if span A creates B and C then they should all have the same name but different "resource"? In our usecase I feel something here is redundant, we only want to assign a name to spans and we don't have distinct paths that we wish to track.

@gbbr we have been reaching out via support for all our comms here, btw. If you ever need a ticket number just ask. I think the most recent one is 213436 but I wouldn't get too caught up on whatever the title of the request is: we're all learning as we're going and we're trying to match the datadog features onto our requirements and there is a mismatch that we have to workaround.

**Comment by fommil** [comment]
I think we got to the bottom of this and experimented with some workarounds in stack.

TL;DR the backend is silently dropping these spans!
